jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -574,25 +580,33 @@ ecma_builtin_typedarray_prototype_filter (ecma_value_t this_arg, /**< this argum
   if (ecma_is_value_empty (ret_value))
   {
     ecma_value_t new_typedarray = ecma_op_create_typedarray_with_type_and_length (obj_p, pass_num);
-    ecma_object_t *new_obj_p = ecma_get_object_from_value (new_typedarray);
-
-    for (uint32_t index = 0; index < pass_num && ecma_is_value_empty (ret_value); index++)
-    {
-      ecma_value_t set_status = ecma_op_typedarray_set_index_prop (new_obj_p, index, *(pass_value_list + index));
-
-      if (!set_status)
-      {
-        ret_value = ecma_raise_type_error (ECMA_ERR_MSG ("error in typedarray set"));
-      }
-    }
 
-    if (ecma_is_value_empty (ret_value))
+    if (ECMA_IS_VALUE_ERROR (new_typedarray))
     {
       ret_value = new_typedarray;
     }
     else
     {
-      ecma_free_value (new_typedarray);
+      ecma_object_t *new_obj_p = ecma_get_object_from_value (new_typedarray);
+
+      for (uint32_t index = 0; index < pass_num && ecma_is_value_empty (ret_value); index++)
+      {
+        ecma_value_t set_status = ecma_op_typedarray_set_index_prop (new_obj_p, index, *(pass_value_list + index));
+
+        if (!set_status)
```

Comment:
Hm, I am not sure I understand this. set_status is an ecma_value but here a !set_status is checked. Is it a bool?